packages/core/src/external/quest/fhir/service-request.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  CodeableConcept,
+  Coding,
+  Identifier,
+  Practitioner,
+  Reference,
+  ServiceRequest,
+} from "@medplum/fhirtypes";
+import { uuidv7 } from "@metriport/shared/util/uuid-v7";
+import { ResponseDetail } from "../schema/response";
+import { getPractitionerReference } from "./practitioner";
+import { getQuestDataSourceExtension } from "./shared";
+
+export function getServiceRequest(
+  detail: ResponseDetail,
+  { requestingPractitioner }: { requestingPractitioner: Practitioner }
+): ServiceRequest {
+  const identifier = getServiceRequestIdentifier(detail);
+  const requester = getPractitionerReference(requestingPractitioner);
+  const code = getServiceRequestCoding(detail);
+  const extension = [getQuestDataSourceExtension()];
+
+  return {
+    resourceType: "ServiceRequest",
+    id: uuidv7(),
+    status: "completed",
+    intent: "original-order",
+    ...(code ? { code } : {}),
+    ...(identifier ? { identifier } : {}),
+    requester,
+    extension,
+  };
```

Comment:
Mapping improvements
1. Could specify category, given we know its a lab, right?
<img width="799" height="270" alt="image" src="https://github.com/user-attachments/assets/ed23aff6-cb17-49ae-8a12-639abbf92160" />
2. subject should point back to the patient!
3. specimen should point back to the specimen! It doesn't map that nicely as its a bidirectional link but def worth doing. references as some of the most key parts of the data as a single edge unlocks tons of new info in the data.
4. locationReference should point to a location created using the orderingAccount address information (i believe, please lmk if I'm wrong based on the data you've seen)

spec: https://hl7.org/fhir/R4/servicerequest.html

Reply:
@lucasdellabella Sorry I just realized I overlooked some of these - checking them carefully:
1. Addressed, created a shared function similar to the other category functions
2. Addressed
3. Addressed, created a separate function to add the reference after the later creation of the specimen
4. Addressed, added location reference

Reply:
@keshavsaharia Leave a comment in getServiceRequest pointing out that we wire in the specimen downstream due to a bidirectional reference (so one needs to be created before the other.

I would consider quickly just making a parent function

`getServiceRequestAndSpecimen` that creates both, wires them together, and returns them.